ReelTalk/index.ios.js

```diff
@@ -1,87 +1,271 @@
 'use strict';
 
-var React = require('react-native');
-var {
+import React, {
   AppRegistry,
-  NavigatorIOS,
   StyleSheet,
   TabBarIOS,
-} = React;
+  Text,
+  Navigator,
+  NavigatorIOS,
+  View,
+  TouchableOpacity,
+  TouchableHighlight
+} from 'react-native';
 
-var RecommendScreen = require('./screens/RecommendScreen');
-var ListsScreen = require('./screens/ListsScreen');
-var TopChartsScreen = require('./screens/TopChartsScreen');
-var SettingsScreen = require('./screens/SettingsScreen');
+import RecommendScreen from './screens/RecommendScreen';
+import ListsScreen from './screens/ListsScreen';
+import TopChartsScreen from './screens/TopChartsScreen';
+import SettingsScreen from './screens/SettingsScreen';
 
-var ReelTalk = React.createClass({
+import cssVar from 'cssVar';
+
+const NavigationBarRouteMapper = {
+  LeftButton: function(route, navigator, index, navState) {
+    if (index === 0) {
+      return null;
+    }
+
+    const previousRoute = navState.routeStack[index - 1];
+    return (
+      <TouchableOpacity
+        onPress={() => navigator.pop()}
+        style={styles.navBarLeftButton}>
+        <Text style={[styles.navBarText, styles.navBarButtonText]}>
+          {"< " + previousRoute.title}
+        </Text>
+      </TouchableOpacity>
+    );
+  },
+
+  RightButton: function(route, navigator, index, navState) {
+    return null;
+  },
+
+  Title: function(route, navigator, index, navState) {
+    return (
+      <Text style={[styles.navBarText, styles.navBarTitleText]}>
+        {route.title}
+      </Text>
+    );
+  },
+
+};
+
+const Main = React.createClass({
+  getInitialState: function() {
+    return { state: this.props.activeTab }
+  },
+
+  componentDidMount: function() {
+    this.refs.recommendTabRef.setState({hasBeenSelected: true})
+  },
+
+  renderRecommendScreen: function() {
```

Comment:
I think we should try and find a way to keep the screens separate so we can reuse them, and maybe pass functions to change the state of the navigator (if that is the reason these are here). Maybe something like this: https://github.com/brentvatne/react-native-navigator-ios-hide-bugfix/commit/e0655d19821072f7d2ace9bea42902ab50e9d861 which originated from here: https://github.com/facebook/react-native/issues/846